WIP: Development

jamillosantos opened the pull request:
...

Jeconias committed:
Refactoring

Jeconias committed:
Corrections in tests and enhancement.

Jeconias committed:
WIP: Add new test for check reference between packages.

Jeconias committed:
WIP: Mapping and testing comments in AST.

Jeconias committed:
:sparkles: Struct parse and enhancement in tests (#1)

Jeconias committed:
:sparkles: Comments parse and enhancement in tests  (#3)

Jeconias committed:
:sparkles: Enable parsing Builtin (#4)

Jeconias committed:
:sparkles: Variables parse (#6)

Jeconias reviewed (approved):
This code review has been focused on parts with few refactoring. Apparently everything is fine.

jamillosantos committed:
:sparkles: Implements `os/build` integration alongside many improvements (#8)

Co-authored-by: Jack <jeconiass2009@hotmail.com>
Co-authored-by: Jota <jamillo@gmail.com>

jamillosantos committed:
:bug: Update the go.mod lib name

jamillosantos committed:
:sparkles: Add method for parsing directories  (#9)

jamillosantos committed:
:sparkles: Implement parsing interfaces (#10)

* :refactor: Split method, interface and struct definitions

* :sparkles: Implement MethodDescriptor.Compatible

This method enables to compare two MethodDescriptor and check if they are
compatible. Method are considered compatatibles if they have the same
signature (only types of arguments and results are considered).

* :sparkles: Implement interface parsing

* :sparkles: Add `token.FSet` the `parseFileContext`

This commit enables the possibility of parseErrors be reported with
file names and line numbers.

* :bug: Composed interfaces broke the parse

This commit do not implement composing interfaces. But, it gracefully
ignores them.

jamillosantos committed:
:arrow_up: Update macchiato

jamillosantos committed:
Implement listeners (#11)

jamillosantos committed:
Fix parsing anonymous interface and structs (#13)

* :bug: Fix struct parsing

Issue: #12

jamillosantos committed:
🐛 Fix test errors

jamillosantos commented on data/models6.sample.go:
```diff
@@ -0,0 +1,13 @@
+}
+
+func show(name string, age int64) {
+	fmt.Println(age, name)
```
A couple of years off, hahaha. The goal of this code is not to work, it is to be parsed.
